lib/system.nim

```diff
@@ -3893,6 +3891,65 @@ proc compiles*(x: untyped): bool {.magic: "Compiles", noSideEffect, compileTime.
 when declared(initDebugger):
   initDebugger()
 
+proc addEscapedChar*(s: var string, c: char) {.noSideEffect, inline.} =
```

Comment:
I would prefer a ``escape(c: char): string``. Would that affect performance too much?

Reply:
Yes, this would allocate a temporary string object for every character to be added. 

The reasoning for using `addEscapeChar` was: `system` provides the rather lower level `add...` functions which are all fast in-place `s: var string` operations, and `strutils` offers the higher level variants. Having `escape(c: char): string` in `system` and another `escape(s: string): string` in `strutils` would feel a bit inconsistent as well.